torchvision/prototype/transforms/_color.py

```diff
@@ -117,14 +116,14 @@ def _permute_channels(self, inpt: Any, *, permutation: torch.Tensor) -> Any:
 
         image = inpt
         if isinstance(inpt, PIL.Image.Image):
-            image = _F.pil_to_tensor(image)
+            image = F.to_image_tensor(image)
```

Comment:
Github won't let me put the comment on the right place but 5 lines above we are doing a `not (isinstance(inpt, (features.Image, PIL.Image.Image)) or is_simple_tensor(inpt))` check to pass through. Instead we should declare in this class a:
`_transformed_types = (features.Image, PIL.Image.Image, is_simple_tensor)`

And avoid this.

Reply:
On a second thought, it might be worth applying this proposal to all Colour kernels to avoid the dispatching to the no-op. This is going to make things faster. Since this is a bigger change, I'll send a separate PR and discuss there if we want to do it.